src/box/lua/config/utils/schema.lua

```diff
@@ -1492,6 +1492,88 @@ end
 
 -- }}} <schema object>:pairs()
 
+-- {{{ <schema object>:jsonschema()
+
+local json_scalars = {
+    string = {jsonschema = {type = 'string'}},
+    number = {jsonschema = {type = 'number'}},
+    ['string, number'] = {jsonschema = {type = {'string', 'number'}}},
+    ['number, string'] = {jsonschema = {type = {'string', 'number'}}},
+    integer = {jsonschema = {type = 'integer'}},
+    boolean = {jsonschema = {type = 'boolean'}},
+    any = {jsonschema = {}}
+}
+
+-- This function sets common fields such as `description`, `default`, and `enum`
+-- for a given schema object. These fields are typical in JSON schemas.
+local function set_common_fields(res, schema)
+    if schema.description ~= nil then
+        res.description = schema.description
+    end
+    res.default = schema.default
+    res.enum = schema.allowed_values
+    return setmetatable(res, {
+        __serialize = 'map',
+    })
+end
+
+local function traverse_impl(schema)
+    local schema_type = schema.type
+
+    if is_scalar(schema) then
+        -- If the schema is a scalar type (string, number, etc.)
+        -- copy the corresponding JSON schema.
+        local scalar_copy = table.copy(json_scalars[schema_type].jsonschema)
+        return set_common_fields(scalar_copy, schema)
+
+    elseif schema_type == 'record' then
+        local properties = {}
+        for field_name, field_def in pairs(schema.fields) do
+            properties[field_name] = traverse_impl(field_def)
+        end
+        return set_common_fields({
+            type = 'object',
+            properties = properties,
+            additionalProperties = false
+        }, schema)
+
+    elseif schema_type == 'map' then
+        assert(schema.key.type == 'string')
```

Comment:
Our schema implementation doesn't forbid non-string keys. JSON has no support for it, so, I guess, JSON Schema too. We should give a user-friendly error in the case, not just an assert.

The same if a record has a field with a non-string name.

Reply:
BTW, if we report an error here, it seems worthful to use `walkthrough_error()`. So, we need to call `walkthrough_enter()` and `walkthrough_leave()`. So, we have to create a context in `:jsonschema()` using `walkthrough_start()` and pass it into the recurive function. See how it is implemented in other methods.

Reply:
I use `walkthrough_*` and set map key name as `<*>` and array index as `[*]`. Is it ok for you? Or we should make it the same as the field_path, using `*` in both cases.

Reply:
I would add `*` for both cases.

Reply:
Done.